automation/src/main/java/org/wso2/testgrid/automation/executor/TestExecutor.java

```diff
@@ -66,11 +69,16 @@ public abstract void init(String testsLocation, String testName, TestScenario te
      * @return The response of script execution.
      * @throws TestAutomationException Throws exception when there is an error executing the script.
      */
-    public String executeEnvironmentScript(Path script, DeploymentCreationResult deploymentCreationResult)
+    public int executeEnvironmentScript(Path script, DeploymentCreationResult deploymentCreationResult)
             throws TestAutomationException {
         try {
-            return TestGridUtil.executeCommand("bash " + script.toString(), null,
-                    deploymentCreationResult);
+            ShellExecutor shellExecutor = new ShellExecutor();
+            Map<String, String> environment = new HashMap<>();
+
+            for (Host host : deploymentCreationResult.getHosts()) {
+                environment.put(host.getLabel(), host.getIp());
```

Comment:
Does host.getLabel method always returns a value? If not, could be happen NPE.

Reply:
Yes, the CF script will output some hosts.